pkg/wamr/Makefile

```
@@ -0,0 +1,81 @@
+PKG_NAME=wamr
+PKG_URL=https://github.com/bytecodealliance/wasm-micro-runtime.git
+PKG_BLEADING?=1
+ifeq ($(PKG_BLEADING),1)
+  PKG_VERSION=main
+else
+  PKG_VERSION=8fd89bd415db73281c8fbd53de1c2726f35b5ddb
+endif
+PKG_LICENSE=Apache-2.0
+
+PKG_CUSTOM_PREPARED = CHECK_VERSION
+
+include $(RIOTBASE)/pkg/pkg.mk
+
+#less Wall TODO: get things fixed upstream
+CFLAGS :=  $(filter-out -pedantic, $(CFLAGS))
+CFLAGS += -Wno-format
+CFLAGS += -Wno-strict-prototypes
+CFLAGS += -Wno-old-style-definition
+CFLAGS += -Wno-cast-function-type
+
+#Prepapre includes for cmake
+RIOT_INCLUDES = $(filter-out -%,$(subst -I,,$(INCLUDES)))
+
+#translate (CPU_ARCH) to Build Target
+#WAMR_BUILD_TARGET is "X86_32" "AARCH64[sub]", "ARM[sub]",
+#                     "THUMB[sub]", "MIPS" or "XTENSA"
+#no msp430, no AVR support for now
+
+ifeq ($(CPU),native)
+#$(CPU) is defined for every CPU
+#everyone build on x86_32
+  WAMR_BUILD_TARGET = X86_32
+else ifeq ($(findstring arm,$(CPU_ARCH)),arm)
+  WAMR_BUILD_TARGET = THUMB
+else ifeq ($(CPU_ARCH),mips32r2)
+  WAMR_BUILD_TARGET = MIPS
+else ifeq ($(CPU_ARCH),xtensa)
+  WAMR_BUILD_TARGET = XTENSA
+else ifeq ($(CPU_ARCH),rv32)
+  WAMR_BUILD_TARGET = RISCV32
+endif
```

```suggestion
else ifeq ($(CPU_ARCH),rv32)
  WAMR_BUILD_TARGET = RISCV32
else
  $(error "WASM is not supported for architecture $(CPU_ARCH)")
endif
```

(I'm aware that this should never trigger with the current state of RIOT's code base, but better be safe than sorry.)